src/v/config/configuration.cc

```diff
@@ -866,6 +866,19 @@ configuration::configuration()
       "limit applies to compressed batch size",
       {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
       1_MiB)
+  , kafka_nodelete_topics(
+      *this,
+      "kafka_nodelete_topics",
+      "Prevents the topics in the list from being deleted via the kafka api",
+      {.needs_restart = needs_restart::no, .visibility = visibility::user},
+      {})
+  , kafka_noproduce_topics(
+      *this,
+      "kafka_noproduce_topics",
+      "Prevents the topics in the list from having message produced to them "
+      "via the kafka api",
+      {.needs_restart = needs_restart::no, .visibility = visibility::user},
+      {})
```

Comment:
The defaults should include the topics we want to protect by default, + extend tests to validate that these topics are protected by default.

Reply:
Will add the default topics mentioned in the RFC for RP and cloud. And extend tests to ensure they are validation. One thing to note is even though the topics will be protected by default they will not be created by default so something will still have to create them.

Reply:
Added every non-cloud topic I saw in the RFC as defaults to the two properties. This will still allow the topics to be created and configured during the initial creation, but will prevent any configuration thereafter.

One change I made over the recommendations in the RFC is that `_schemas` was only added to the nodelete property and not the noproduce property. This change was made after consulting with @BenPope . It looks like the `_schemas` topic needs to be able to be produced to by the Kafka API.